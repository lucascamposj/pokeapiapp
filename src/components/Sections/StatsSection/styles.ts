import styled from 'styled-components/native';

export const Container = styled.View`
  /* flex: 1; */
  flex-direction: row;
  align-content: center;
  flex-wrap: wrap;
  justify-content: center;
`;
