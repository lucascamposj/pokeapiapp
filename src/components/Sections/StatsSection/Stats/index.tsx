import React from 'react';
import { Container, StatText, StatsContent, StatNumber } from './styles';
import { StatInterface } from '../../../../pages/Pokemon';

const Stat: React.FC<StatInterface> = ({
  stat,
  base_stat,
  effort,
}: StatInterface) => {
  return (
    <Container>
      <StatText>{stat.name}</StatText>
      <StatsContent>
        <StatNumber>{base_stat}</StatNumber>
        <StatNumber>{effort && ` (${effort})`}</StatNumber>
      </StatsContent>
    </Container>
  );
};

export default Stat;
