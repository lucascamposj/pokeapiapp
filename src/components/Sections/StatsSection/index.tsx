import React from 'react';
import Stat from './Stats';
import { Container } from './styles';
import { StatInterface } from '../../../pages/Pokemon';

interface StatSectionProps {
  stats: StatInterface[] | undefined;
}

const StatSection: React.FC<StatSectionProps> = ({
  stats,
}: StatSectionProps) => {
  return (
    <Container>
      {stats?.map(({ stat, base_stat, effort }) => (
        <Stat
          key={stat.name}
          stat={stat}
          base_stat={base_stat}
          effort={effort}
        />
      ))}
    </Container>
  );
};

export default StatSection;
