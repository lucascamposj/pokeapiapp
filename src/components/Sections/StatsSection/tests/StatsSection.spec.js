import React from 'react';
import {shallow} from 'enzyme';
import StatsSection from '../../StatsSection';

describe('StatsSection', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const stats = [
              {
                stat: {
                  name: "HP",
                  effort: 0,
                  base_stat: 450
                }
              },
              {
                stat: {
                  name: "ATTACK",
                  effort: 2,
                  base_stat: 490
                }
              }
            ]
            const component = shallow(
              <StatsSection
                stats={stats}
              />)
            expect(component).toMatchSnapshot()
        });
    });
});
