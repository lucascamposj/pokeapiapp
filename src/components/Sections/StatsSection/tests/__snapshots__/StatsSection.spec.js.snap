// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsSection Rendering should match to snapshot 1`] = `
<Styled(View)>
  <Stat
    key="HP"
    stat={
      Object {
        "base_stat": 450,
        "effort": 0,
        "name": "HP",
      }
    }
  />
  <Stat
    key="ATTACK"
    stat={
      Object {
        "base_stat": 490,
        "effort": 2,
        "name": "ATTACK",
      }
    }
  />
</Styled(View)>
`;
